_posts/2020-09-17-accessing-snowplow-data-in-real-time-aws/index.md

```diff
@@ -0,0 +1,123 @@
+
+```python
+import snowplow_analytics_sdk.event_transformer
+import snowplow_analytics_sdk.snowplow_event_transformation_exception
+import json
+import base64
+
+# Decode batch of lambda records from Base64 to TSV
+def decode_records(update):
+    data = []
+    if "Records" in update:
+        for record in update["Records"]:
+            if "kinesis" in record and "data" in record['kinesis']:
+                # decode base64 data in each record and append to the list
+                decoded_data = base64.b64decode(record['kinesis']['data']).decode('utf-8')
+                data.append(decoded_data)
+    return data
+
+# Entry point for the lambda function
+def lambda_handler(event, context):
+    # Uncomment this line to show the raw batch of events
+    # print("Received event: " + json.dumps(event, indent=2)) 
+
+    # Decode received batch of records from Base64 to Snowplow TSV
+    records = decode_records(event)
+
+    for record in records:
+        try:
+            # Transform record from Snowplow TSV to JSON
+            print(snowplow_analytics_sdk.event_transformer.transform(record))
+            print("Succesfully retrieved and transformed event")
+        except snowplow_analytics_sdk.snowplow_event_transformation_exception.SnowplowEventTransformationException as e:
+            for error_message in e.error_messages:
+                print(error_message)
+            continue
+        
+    return
+```
+
+## 5. Package and upload the Python script
+Package a zip file for lambda with the following script. Copy it to a new file and place it in the directory above the python script. Note that this is being run on a subdirectry with the Python app called lambda_function_payload.py. The Python fuction above is written in Python 3 the script below will install the [Snowplow Python Analytics SDK](https://github.com/snowplow/snowplow/wiki/Python-Analytics-SDK-Setup) in the directory so that it is included in the zip file.
+
+```bash
+#!/bin/bash
+cd lambda_function_payload
+pip3 install snowplow_analytics_sdk -t . --upgrade
+chmod -R 755 .
+zip -r ../lambda_function_payload.zip .
+cd ..
+```
+
+Upload the zip file to lambda in the AWS Console. Since the dependency is a couple of mb, we won't be able to see and edit the code in AWS. 
+![Screenshot](img/upload_zip.png)
+
+Set up the basic settings to point to the handler in the Python script. The handler is the entry point into the code where execution will begin. The handler is formated as`name_of_python_file.name_of_handler_function`. The `.py` is not included.
+![Screenshot](img/basic_settings.png)
+
+## 6. Add a trigger to connect it to the enriched kinesis stream
+Set up your trigger based on the screenshot below. The default values will be fine but note that your Kinesis stream will be named differently. 
```

Comment:
```suggestion
Set up your trigger based on the screenshot below. The default values will be fine but note that your Kinesis stream will be named differently. For your production workloads, you may wish to use different batch sizes and utilise retries but this will vary depending on your use case.  

```